src/v/s3/client.cc

```diff
@@ -693,7 +827,19 @@ ss::future<client::list_bucket_result> client::list_objects_v2(
       });
 }
 
-ss::future<> client::delete_object(
+ss::future<result<client::no_response, error_outcome>> client::delete_object(
+  const bucket_name& bucket,
+  const object_key& key,
+  const ss::lowres_clock::duration& timeout) {
+    return send_request(
```

Comment:
Should we map `not_found` to `none` here to avoid `remote` having to understand the quirk of some backends returning 404 sometimes?

Reply:
Good idea. We should also move the current log for this scenario to `trace`.

Reply:
Done in https://github.com/redpanda-data/redpanda/pull/7706/commits/ca54f0c56c87764ff9f2a1c50063c48229a46ea2. I mapped to `no_response` instead of `none` to keep error handling consistent.